csharp/tools/MauiModelTester/MainPage.xaml.cs

```diff
@@ -0,0 +1,173 @@
+            nameof(ExecutionProviders.NNAPI) => ExecutionProviders.NNAPI,
+            nameof(ExecutionProviders.CoreML) => ExecutionProviders.CoreML,
+            nameof(ExecutionProviders.XNNPACK) => ExecutionProviders.XNNPACK,
+            _ => ExecutionProviders.CPU
+        };
+
+        if (_inferenceSession == null || executionProvider != _currentExecutionProvider)
+        {
+            _currentExecutionProvider = executionProvider;
+
+            // re/create an inference session with the execution provider.
+            // this is an expensive operation as we have to reload the model, and should be avoided in production apps.
+            _inferenceSession = new OrtInferenceSession(_currentExecutionProvider);
+            await _inferenceSession.Create();
+
+            // Display the results which at this point will have the model load time and the warmup Run() time.
+            ShowResults();
+        }
+    }
+
+    private void ExecutionProviderOptions_SelectedIndexChanged(object sender, EventArgs e)
+    {
+        UpdateExecutionProvider().ContinueWith((task) =>
+                                               {
+                                                   MainThread.BeginInvokeOnMainThread(
+                                                       () =>
+                                                       {
+                                                           if (task.IsFaulted)
+                                                           {
+                                                               DisplayAlert("Error", task.Exception.Message, "OK");
+                                                           }
+                                                       });
+                                               });
+    }
+
+    private void OnRunClicked(object sender, EventArgs e)
+    {
+        RunAsync().ContinueWith(resultTask =>
+                                {
+                                    if (resultTask.IsFaulted)
+                                    {
+                                        MainThread.BeginInvokeOnMainThread(
+                                            () => DisplayAlert("Error", resultTask.Exception.Message, "OK"));
+                                    }
+                                });
+    }
+
+    private async Task UpdateExecutionProvider()
+    {
+        await SetBusy(true);
+        await CreateInferenceSession();
+        await SetBusy(false);
+    }
+
+    private async Task RunAsync()
+    {
+        await ClearResult();
+
+        var iterationsStr = Iterations.Text;
+        int iterations = iterationsStr == string.Empty ? 10 : int.Parse(iterationsStr);
+
+        await SetBusy(true);
+
+        // create inference session if it doesn't exist or EP has changed
+        await CreateInferenceSession();
+
+        await Task.Run(() => _inferenceSession.Run(iterations));
+
+        await SetBusy(false);
```

Comment:
do we need to call `SetBusy(false)` even when an exception is thrown? otherwise the UI could get stuck in the busy state. if we handle exceptions in this method (and `UpdateExecutionProvider`), maybe we can move the error handling from the `ContinueWith` to a `catch` here.